Amazon.IonDotnet.Tests/Internals/TreeReaderTest.cs

```diff
@@ -244,6 +244,17 @@ public void ValueWithTypeAnnotationsTest()
             ReaderTestCommon.ReadTypeAnnotations_SingleField(reader);
         }
 
+        [TestMethod]
+        public void ValueWithTypeAnnotationsTest_ZeroSymbol()
+        {
+            //Must be: $0::18
+            var intValue = _ionValueFactory.NewInt(18);
+            intValue.AddTypeAnnotation(null);
```

Comment:
`AddTypeAnnotation(null)` should throw an exception. To add symbol zero as an annotation, the user must provide a `SymbolToken`.

Reply:
Updated to AddTypeAnnotation(SymbolToken)

Reply:
Also added null check in AddTypeAnnotation(String) and added unit test